nixos/modules/services/security/openbao.nix

```diff
@@ -0,0 +1,140 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.services.openbao;
+
+  settingsFormat = pkgs.formats.json { };
+in
+{
+  options = {
+    services.openbao = {
+      enable = lib.mkEnableOption "OpenBao daemon";
+
+      package = lib.mkPackageOption pkgs "openbao" { };
+
+      settings = lib.mkOption {
+        description = ''
+          Settings of OpenBao.
+
+          See [documentation](https://openbao.org/docs/configuration) for more details.
+        '';
+        example = lib.literalExpression ''
+          {
+            listener.tcp = {
+              tls_acme_email = config.security.acme.defaults.email;
+              tls_acme_domains = [ "example.com" ];
+              tls_acme_disable_http_challenge = true;
+            };
+
+            cluster_addr = "http://127.0.0.1:8201";
+            api_addr = "https://example.com";
+
+            storage.raft = {
+              path = "/var/lib/openbao";
+              node_id = config.networking.hostName;
+            };
+          }
+        '';
+        type = lib.types.submodule {
+          freeformType = settingsFormat.type;
+          options = {
+            listener.tcp = {
+              address = lib.mkOption {
+                type = lib.types.str;
+                default = "127.0.0.1:8200";
+                description = ''
+                  Specifies the address to bind to for listening.
+                '';
+              };
+
+              cluster_address = lib.mkOption {
+                type = lib.types.str;
+                default = "127.0.0.1:8201";
```

Comment:
imho this default doesn't make much sense - cluster_addr must be reachable from other nodes so one would have to declare it in any case if they want a functional cluster. I'd recommend either to drop it completely or replace it with `0.0.0.0:8201`.

Reply:
I copied the defaults from openbao's docs.

Reply:
I got rid of this option for now.